sourcegraph/sourcegraph
# svelte: Remove full page scrolling of repo pages

This commit removes the full page scrolling behavior for repo root, directory and file pages. The implementation had several issues.

In the new implementation the scroll container is CodeMirror itself. It provides the following behavior:

- Show the top of the file when navigating to a new file.
- Scroll the selected line into view when opening a URL containing a position.
- Do not scroll when selecting a line (currently broken).
- Restore previous scroll position when navigating backward or forward in the browser history.
- Toggling the blame view maintains the current line view (well, almost but that's as best as we can do at the moment).

Additional changes in this commit:

- Removed logic to enable/disable full page scrolling.
- Remove CSS for making elements `sticky`.
- Update "scroll selected line into view" logic to prevent scrolling when restoring the previous scroll position.
- Update `codemirror/view` to include a recent fix for a scroll related bug.

NOTE: Due to these changes we lost scroll restoration for repo root and directory pages. This will be fixed in a separate PR.

## Test plan

`pnpm test` passes

Manual testing:

- Open file page in new tab -> file is scrolled to the top
- Open file page with selected line in new tab -> selected line is scrolled into view
- Navigate to new file via file tree -> new file is scrolled to the top
- Scroll file, navigate to new file and back -> scroll position is restored
- Scroll file, navigate back, navigate forward -> scroll position is restored

- Preview panel appears to be working (can scroll and navigate between matches)
- Directory page is scrollable
- Repo root page is scrollable
